addons/project/models/project.py

```diff
@@ -456,6 +457,12 @@ def map_tasks(self, new_project_id):
             new_child_ids = [old_to_new_tasks[child.id] for child in task.child_ids if child.id in old_to_new_tasks]
             tasks.browse(new_child_ids).write({'parent_id': new_task.id})
             old_to_new_tasks[task.id] = new_task.id
+            depend_on_ids = [t.id if t not in all_tasks else old_to_new_tasks.get(t.id, None) for t in
+                             task.depend_on_ids]
+            new_task.write({'depend_on_ids': [(4, tid) for tid in depend_on_ids if tid]})
```

Comment:
You can use the Command class instead of using the magic numbers. 🙂 
```suggestion
            new_task.write({'depend_on_ids': [Command.link(tid) for tid in depend_on_ids if tid]})
```